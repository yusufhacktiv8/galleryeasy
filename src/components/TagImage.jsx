import React from 'react';

const TAG_IMAGE_URL = '/images/tagimage.png';

export default ({ transparent, onClick }) => {
  const className = transparent ? 'tag-img transparent-70' : 'tag-img';
  return (
    <img src={TAG_IMAGE_URL} className={className} onClick={onClick} alt="Tag" />
  );
};
