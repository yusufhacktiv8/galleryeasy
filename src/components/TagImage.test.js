import React from 'react';
import TagImage from './TagImage';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

test('Default TagImage correctly rendered', () => {
  const component = renderer.create(
    <TagImage />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Transparent TagImage correctly rendered', () => {
  const component = renderer.create(
    <TagImage transparent />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('TagImage onClick called when clicked', () => {
  const callback = sinon.spy();
  const component = renderer.create(
    <TagImage onClick={callback} />,
  );
  let tree = component.toJSON();
  tree.props.onClick();
  
  expect(callback.called).toBe(true);
});
