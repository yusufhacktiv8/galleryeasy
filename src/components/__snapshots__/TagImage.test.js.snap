// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default TagImage correctly rendered 1`] = `
<img
  alt="Tag"
  className="tag-img"
  onClick={undefined}
  src="/images/tagimage.png"
/>
`;

exports[`Transparent TagImage correctly rendered 1`] = `
<img
  alt="Tag"
  className="tag-img transparent-70"
  onClick={undefined}
  src="/images/tagimage.png"
/>
`;
